golem-api-grpc/proto/golem/worker/invocation_context.proto

```diff
@@ -2,10 +2,41 @@ syntax = "proto3";
 
 package golem.worker;
 
+import "google/protobuf/timestamp.proto";
 import "golem/worker/worker_id.proto";
 
 message InvocationContext {
   golem.worker.WorkerId parent = 1;
   repeated string args = 3;
   map<string, string> env = 4;
+  optional TracingInvocationContext tracing = 5;
+}
+
+message TracingInvocationContext {
+  string trace_id = 1;               // hexadecimal representation of a 128-bit trace ID
+  repeated InvocationSpan spans = 2; // stack of spans, first item is the current, last one is the root
```

Comment:
What are the cases in which we would begin an invocation with more than one local stack?

I'm thinking we could have this be closer to the opentelemetry standard and just have ids for parents on each levels instead of having this stack of spans https://opentelemetry.io/docs/concepts/signals/traces/.

Reply:
See the corresponding notion doc where we decided that we want the workers to have access to the whole context stack. If we would just send ids, then we would still be able to export spans to opentelemetry but we would not be able to query information from the workers.

Reply:
Well they would only be able to query the current span, which I think is inline with how other opentelemetry setups work.

But makes sense if we want to support getting the full stack. In that case, should there ever be a remote parent? As users cannot access that one from within the component (at least not without some new host api)